theano/compile/mode.py

```diff
@@ -323,6 +319,30 @@ def including(self, *tags):
         #     string? Optimizer? OptDB? who knows???
         return self.__class__(linker=link, optimizer=opt.including(*tags))
 
+    def register(self, *optimizations):
+        """Adds new optimization instances to a mode.
+
+        This method adds new optimization instances to a compilation mode. It
+        works like the `including()` method but takes as inputs optimization
+        instances to add instead of tags.
+
+        Parameters
+        ----------
+        optimizations :
+            Every element of `optimizations` is a tuple containing an
+            optimization instance and a floating point value indicating the
+            position at which to insert the optimization in the mode.
+
+        Returns
+        -------
+        Mode
+            Copy of the current Mode which includes the provided
+            optimizations.
+        """
+
+        new_optimizer = self.provided_optimizer.register(*optimizations)
```

Comment:
I'm pretty sure provided_optimizer can be a string which does not have the register() method.  You should probably use get_linker_optimizer()